x/stake/keeper.go

```diff
@@ -469,3 +469,67 @@ func (k Keeper) setPool(ctx sdk.Context, p Pool) {
 	store.Set(PoolKey, b)
 	k.gs = Pool{} // clear the cache
 }
+
+//__________________________________________________________________________
+
+// Implements ValidatorSetKeeper
+
+var _ sdk.ValidatorSetKeeper = Keeper{}
+
+func (k Keeper) Validators(ctx sdk.Context) []*sdk.Validator {
+	vals := k.GetValidators(ctx)
+	res := make([]*sdk.Validator, len(vals))
+
+	for i, val := range vals {
+		abcival := val.abciValidator(k.cdc)
+		res[i] = &abcival
+	}
+
+	return res
+}
+
+func (k Keeper) Size(ctx sdk.Context) int {
```

Comment:
The size should be documented in the pool and saved when the validator set is created, we should not have to retrieve the entire validator set for this

Reply:
Removed since `GetByIndex` is removed